src/Microsoft.ML.Transforms/UngroupTransform.cs

```diff
@@ -462,27 +452,27 @@ private sealed class Cursor : LinkedRootCursorBase
             // Parallel to columns.
             private int[] _colSizes;
 
-            public Cursor(IChannelProvider provider, RowCursor input, SchemaImpl schema, Func<int, bool> predicate)
+            public Cursor(IChannelProvider provider, RowCursor input, UngroupBinding schema, Func<int, bool> predicate)
                 : base(provider, input)
             {
-                _schemaImpl = schema;
-                _active = Utils.BuildArray(_schemaImpl.ColumnCount, predicate);
-                _cachedGetters = new Delegate[_schemaImpl.ColumnCount];
-                _colSizes = new int[_schemaImpl.ColumnCount];
+                _ungroupBinding = schema;
+                _active = Utils.BuildArray(_ungroupBinding.OutputSchema.Count, predicate);
```

Comment:
>Count [](start = 72, length = 5)

would this be the cardinality of the input schema, as well?

Reply:
Input and output schemas contain the same amounts of columns but I will change it to `_ungroupBinding.InputColumnCount` for the sake of clarity.

---
In reply to: [246129434](https://github.com/dotnet/machinelearning/pull/2066#discussion_r246129434) [](ancestors = 246129434)